meson.build

```
@@ -89,13 +89,9 @@ foreach arg: test_c_args
 endforeach
 add_project_arguments(project_c_args, language: 'c')
 
-_platforms = []
 host_system = host_machine.system()
-if host_machine.system() == 'windows'
-    _platforms = ['windows']
-endif
-with_platform_windows = _platforms.contains('windows')
-with_platform_darwin = ['darwin', 'ios'].contains(host_system)
+with_platform_windows = host_system == 'windows'
+with_platform_darwin = host_system == 'darwin'
```

This change is not correct, according to [the `system` table](https://mesonbuild.com/Reference-tables.html#operating-system-names) iOS *is* Darwin.

The correct way to check if Meson's building for iOS would look like this:

```suggestion
with_platform_darwin = cc.compiles('''#include <Availability.h>
#include <TargetConditionals.h>
#if !TARGET_OS_OSX || __MAC_OS_X_VERSION_MIN_REQUIRED <= 100100L
#error "Not building for macOS"
#endif''')
```

---

I considered that already, and the correct way would be to use `host_machine.subsystem() == 'macos'`, [but that's only available starting meson 1.2.0](https://mesonbuild.com/Reference-tables.html#subsystem-names-since-120).

---

So instead of bumping meson version, it would be good to add the compile check. Could your please update your PR?